pkg/client/proxy.go

```diff
@@ -108,8 +109,14 @@ func getContextWithGRPCTimeout(parent context.Context) context.Context {
 	return ctx
 }
 
-func getContextWithGRPCLongTimeout(parent context.Context) context.Context {
-	ctx, _ := context.WithTimeout(parent, GRPCServiceLongTimeout)
+// getContextWithGRPCLongTimeout returns a context with given grpcTimeoutSeconds + GRPCServiceTimeout timeout.
+// If grpcTimeoutSeconds is 0, use the default GRPCServiceLongTimeout instead.
+func getContextWithGRPCLongTimeout(parent context.Context, grpcTimeoutSeconds int64) context.Context {
+	grpcTimeout := GRPCServiceLongTimeout
+	if grpcTimeoutSeconds > 0 {
+		grpcTimeout = (time.Second * time.Duration(grpcTimeoutSeconds)) + GRPCServiceTimeout
```

Comment:
Can we use `grpcTimeout = max(grpcTimeoutSeconds, GRPCServiceTimeout)` instead?

Reply:
It looks like this is here to mimic:

https://github.com/longhorn/longhorn-instance-manager/blob/0c7a73712efea62b9a465983f2ec0d353d6c0bd4/pkg/client/proxy.go#L101-L104

The context for this is https://github.com/longhorn/longhorn/issues/4096. I am not sure I understand the details, but it seems it is intentional for the timeout here to be longer than the desired timeout at the engine layer.

Reply:
@ejweber is correct, we want to have slight bigger timeout in the proxy client side comparing to the actual timeout in the engine/replica because proxy server add some delay to the flow

Reply:
I see. @PhanLe1010 Can you leave a comment here for better understanding the context? Thanks.

Reply:
Surely. Fixed. Thank you